Fix TypeError on assign trackDay from schedule JSON

wevtimoteo opened the pull request:
## Motivation

When accessing website I was getting this exception (Ubuntu Chrome Version 86.0.4240.198):

```js
Uncaught (in promise) TypeError: Cannot read property 'querySelector' of null
    at partials.js:41
    at Array.map (<anonymous>)
    at partials.js:38
(anonymous) @ partials.js:41
(anonymous) @ partials.js:38
Promise.then (async)
(anonymous) @ partials.js:37
```

It seems this caused due `schedule` JSON promise being fetched before `schedule` markup in some cases.

## Proposed solution

Call `renderSchedule` function after `schedule` markup being rendered.

## Screenshot

![image](https://user-images.githubusercontent.com/645452/99811311-c3936a80-2b23-11eb-9251-4847385531de.png)

wevtimoteo committed:
Fix TypeError on assign trackDay from schedule JSON

When accessing website I was getting this exception due backticks usage:

```
Uncaught (in promise) TypeError: Cannot read property 'querySelector' of null
    at partials.js:41
    at Array.map (<anonymous>)
    at partials.js:38
(anonymous) @ partials.js:41
(anonymous) @ partials.js:38
Promise.then (async)
(anonymous) @ partials.js:37
```

camilacampos commented on js/partials.js:
```diff
@@ -36,7 +36,7 @@ fetch("data/schedule.json")
 .then(schedule => {
   schedule.map(track => {
-    const trackDay = document.querySelector(`.track[data-day="${track.day}"]`)
+    const trackDay = document.querySelector('.track[data-day="${track.day}"]')
```
Com essa modificação, e servindo o site com o jekyll, eu recebo o seguinte erro:
```js
partials.js:41 Uncaught (in promise) TypeError: Cannot read property 'querySelector' of null
    at partials.js:41
    at Array.map (<anonymous>)
    at partials.js:38
```

wevtimoteo replied:
Eu estou recebendo essa mesma exception em [prod](https://ruby.com.br/) ora renderiza e ora não renderiza :(

Parece que está intermitente, pode ser alguma coisa relacionada ao promise, então esse fix do backticks não resolve o problema.

@camilacampos Existe algum branch com o site em Jekyll?

camilacampos replied:
nops, eu que dei um jekyll serve aqui pra conseguir rodar o projeto na minha máquina haha

wevtimoteo replied:
Ah certo! Testei esse meu fix com o jekyll e tenho a mesma exception que você. Aproveitei para testar no branch `main` com o `jekyll` e o comportamento é o mesmo de prod (intermitente). Acho que precisamos investigar mais a fundo, sugestões?

wevtimoteo replied:
O uso do `backticks` aqui está certo, passou batido a interpolação do `${track.day}`. Estou reproduzindo aqui com o backticks para ver o que pode ser o problema :)

wevtimoteo replied:
@camilacampos descobri o que era: algumas vezes a promise do schedule `JSON` termina antes que a promise `schedule.html`. Então na hora de procurar pelo seletor `.track[data-day="${track.day}"]`, nada é encontrado, porque ainda não renderizou. Acho que nesse caso podemos colocar mais um `.then` na promise do `schedule.html` para aí sim fazer o fetch do schedule JSON. O que acha?

wevtimoteo replied:
Fiz um pseudo-fix em a0d248d, se quiser dar uma revisada/testada :)

camilacampos replied:
Boooa!

wevtimoteo committed:
Make schedule rendering dependent from schedule markup

Sometimes schedule JSON is being fetched before schedule markup
renderization which causes `TypeError` exception.

Roriz commented on js/partials.js:
```diff
@@ -1,3 +1,45 @@
+      return response.json()
+    })
+    .then(schedule => {
+      schedule.map(track => {
```
```suggestion
      schedule.forEach(track => {
```
Não temos nenhum return nessa função

wevtimoteo replied:
Opa @Roriz, como esse código não é novo, vou evitar fazer qualquer modificação nele. Seria legal ver com o pessoal de abrir um PR separado para mudar essa implementação.

camilacampos replied:
@Roriz pode abrir um PR com essas modificações?

Roriz commented on js/partials.js:
```diff
@@ -1,3 +1,45 @@
+
+        const table = trackDay.querySelector('tbody')
+        track.talks.forEach(talk => {
+          name = talk.name ? `<br /> ${talk.name}` : ''
+
+          table.innerHTML +=
+            `
+        <tr class="talk-info">
+        <td>
+          <span title="${talk.hour}"><strong>${talk.hour}</strong></span>
+        </td>
+        <td>
+          <span class="title" title="${talk.title}"><strong>${talk.title}</strong></span>
+          <span class="speaker-name" title="${name}">${name}</span>
```
![image](https://user-images.githubusercontent.com/1077589/99831077-5391de00-2b3d-11eb-8cd1-86db9a93bd23.png)

```suggestion
          table.innerHTML +=
            `
        <tr class="talk-info">
        <td>
          <span title="${talk.hour}"><strong>${talk.hour}</strong></span>
        </td>
        <td>
          <span class="title" title="${talk.title}"><strong>${talk.title}</strong></span>
          <span class="speaker-name" title="${talk.name}"><br /> ${talk.name}</span>
```

wevtimoteo replied:
Mesmo caso que acima :)

camilacampos reviewed (approved):
topper! Valeuzão <3

debborafernandess commented:
Valeuzaço @wevtimoteo !!!!!$
